x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/rest/action/oauth2/RestGetTokenActionTests.java

```diff
@@ -85,7 +89,7 @@ public void sendResponse(RestResponse restResponse) {
         assertThat(map, hasEntry("expires_in", Math.toIntExact(createTokenResponse.getExpiresIn().seconds())));
         assertThat(map, hasEntry("refresh_token", createTokenResponse.getRefreshToken()));
         assertThat(map, hasEntry("kerberos_authentication_response_token", createTokenResponse.getKerberosAuthenticationResponseToken()));
-        assertEquals(5, map.size());
+        assertEquals(6, map.size());
```

Comment:
Along the same logic of existing code, I think we need test the map hasEntry for `authentication` as well.

Reply:
I added a simplified testing here. Note that authentication is an object and response representation of it is structurally not exactly the same (user fields are flattened for instance). We can add validation for each field, but I am not sure it is really necessary. I verify the authentication is there and the user name is the same. Let me know if you think I need to add all fields verification though.